src/mimefactory.rs

```diff
@@ -689,19 +689,16 @@ impl<'a> MimeFactory<'a> {
                 .unwrap(),
             );
 
-            headers.unprotected.push(
+            headers.unprotected.insert(
+                0,
                 Header::new_with_value(
                     "From".into(),
                     vec![Address::new_mailbox(self.from_addr.clone())],
                 )
                 .unwrap(),
             );
         } else {
-            // Unencrypted emails have display names in the unprotected headers
-            headers
-                .unprotected
-                .push(Header::new_with_value("To".into(), to.clone()).unwrap());
-            headers.unprotected.push(from_header);
+            headers.unprotected.insert(0, from_header);
```

Comment:
But if it's already in the protected headers, why do we need to put it to the unprotected also? Is this needed for signed-only messages?

Reply:
Yes, 1 test fails if it is not there